password.go

```diff
@@ -17,9 +20,49 @@ var PasswordQuestionTemplate = `
 {{- color "green+hb"}}? {{color "reset"}}
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}`
 
+type filteredReadlinePasswordOutput struct {
+	len int
+}
+
+// Write will be called by readline to echo the masked characters to the console.  It assumes
+// full control of the output with prompting, but since survey is printing the prompt readline
+// does not play nice here.  Write is called with clear-line statements which need to be
+// ignored then it is called with the full buffered output. So if you type "hi" this is the
+// sequence that will be sent to Write
+// 0: "\033[J\033[2K\r"
+// 1: " \b"
+// 2: "\033[J\033[2K\r"
+// 3: "*" (this is the masked "h" from "hi")
+// 4: "\033[J\033[2K\r"
+// 5: "**" (this is the masked "hi" from "hi")
+func (o *filteredReadlinePasswordOutput) Write(p []byte) (n int, err error) {
```

Comment:
There you go again coming up with ideas that make me question the interface. 

I really like this and it feels like something that a question should be concerned with instead of a prompt but i don't really like the idea of passing `helpText` to `Prompt` although it would closer resemble the signature of `Cleanup`. 

Is this something you imagine on multi-selects too? I'm not sure how one would collect a request for help in the various multi-line prompts. In the case of non-`Input` prompts, i'm not sure how I feel about showing help text based on an input the user cannot see they typed. FWIW, Inquirer somewhat addresses in multi-line prompts by showing a short help text describing the arrow key behaviour until you press one.